### PR TITLE
SQLParser not included in the release archive #4453

### PR DESCRIPTION
**Pull request by aleks-f**

### CODE REVIEW

File: Data/include/Poco/Data/SQLParser.h
Comment (matejk) [inline Data/include/Poco/Data/SQLParser.h:1]: This file can be removed.

File: PocoDoc/cfg/mkdoc-poco.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
 				${PocoBuild}/*/include/Poco/*/*/*.h
 				${PocoBuild}/*/include/Poco/*/*.h
 				${PocoBuild}/*/*/include/Poco/*/*/*.h
-				${PocoBuild}/Data/src/sql-parser/*.h
+				${PocoBuild}/Data/SQLParser/*.h
```
Comment (matejk) [inline PocoDoc/cfg/mkdoc-poco.xml:10]: Is it necessary to have SQL parser includes for user documentation generation?

File: Data/ODBC/Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ objects = Binder ConnectionHandle Connector EnvironmentHandle \
 
 target_includes = $(POCO_BASE)/Data/testsuite/include
 ifndef POCO_DATA_NO_SQL_PARSER
-	target_includes += $(POCO_BASE)/Data/src
+	target_includes += $(POCO_BASE)/Data/SQLParser $(POCO_BASE)/Data/SQLParser/src
```
Comment (matejk) [inline Data/ODBC/Makefile:17]: No need to add SQL parser includes here.

**Review by matejk — CHANGES_REQUESTED** [review]
Some simplifications/cleanup can be done. See other comments.

File: Data/include/Poco/Data/SQLParser.h
Comment (aleks-f) [reply Data/include/Poco/Data/SQLParser.h:1]: It is [used](https://github.com/pocoproject/poco/blob/a3ced88384be4e070a26c397b39281600ded3337/Data/testsuite/src/SQLParserTest.cpp#L18) by testsuite.

Comment (aleks-f) [reply Data/include/Poco/Data/SQLParser.h:1]: actually, it could be removed, probably best to do s

File: PocoDoc/cfg/mkdoc-poco.xml (hunk 1 above)
Comment (aleks-f) [reply PocoDoc/cfg/mkdoc-poco.xml:10]: Probably best to remove it too

**Review by matejk — APPROVED** [review]

**Pull request merged**